File: server/pb/groupshare/data/data.op.go
package data

import (
	"github.com/dave/groupshare/server/pb/groupshare/data/foo"
	"github.com/dave/protod/delta"
)

func Op() Op_root_type {
	return Op_root_type{}
}

type Op_root_type struct{}

func (Op_root_type) Share() Share_type {
	return Share_type{}
}

type Share_type struct {
	location []*delta.Locator
}

func (b Share_type) Location_get() []*delta.Locator {
	return b.location
}
func NewShare_type(l []*delta.Locator) Share_type {
	return Share_type{location: l}
}
func (b Share_type) Name() delta.String_scalar {
	return delta.NewString_scalar(delta.CopyAndAppendField(b.location, "name", 1))
}
func (b Share_type) Foo() foo.Foo_type {
	return foo.NewFoo_type(delta.CopyAndAppendField(b.location, "foo", 2))
}
func (b Share_type) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Share_type) Set(value *Share) *delta.Op {
	return delta.Set(b.location, value)
}

type Share_type_list struct {
	location []*delta.Locator
}

func (b Share_type_list) Location_get() []*delta.Locator {
	return b.location
}
func NewShare_type_list(l []*delta.Locator) Share_type_list {
	return Share_type_list{location: l}
}
func (b Share_type_list) Index(i int) Share_type {
	return NewShare_type(delta.CopyAndAppendIndex(b.location, int64(i)))
}
func (b Share_type_list) Insert(index int, value *Share) *delta.Op {
	return delta.Insert(delta.CopyAndAppendIndex(b.location, int64(index)), value)
}
func (b Share_type_list) Move(from, to int) *delta.Op {
	return delta.Move(delta.CopyAndAppendIndex(b.location, int64(from)), int64(to))
}
func (b Share_type_list) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Share_type_list) Set(value []*Share) *delta.Op {
	return delta.Set(b.location, value)
}

type Share_type_bool_map struct {
	location []*delta.Locator
}

func (b Share_type_bool_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShare_type_bool_map(l []*delta.Locator) Share_type_bool_map {
	return Share_type_bool_map{location: l}
}
func (b Share_type_bool_map) Key(key bool) Share_type {
	return NewShare_type(delta.CopyAndAppendKeyBool(b.location, key))
}
func (b Share_type_bool_map) Rename(from, to bool) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyBool(b.location, from), to)
}
func (b Share_type_bool_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Share_type_bool_map) Set(value map[bool]*Share) *delta.Op {
	return delta.Set(b.location, value)
}

type Share_type_int32_map struct {
	location []*delta.Locator
}

func (b Share_type_int32_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShare_type_int32_map(l []*delta.Locator) Share_type_int32_map {
	return Share_type_int32_map{location: l}
}
func (b Share_type_int32_map) Key(key int) Share_type {
	return NewShare_type(delta.CopyAndAppendKeyInt32(b.location, int32(key)))
}
func (b Share_type_int32_map) Rename(from, to int) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyInt32(b.location, int32(from)), int32(to))
}
func (b Share_type_int32_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Share_type_int32_map) Set(value map[int32]*Share) *delta.Op {
	return delta.Set(b.location, value)
}

type Share_type_int64_map struct {
	location []*delta.Locator
}

func (b Share_type_int64_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShare_type_int64_map(l []*delta.Locator) Share_type_int64_map {
	return Share_type_int64_map{location: l}
}
func (b Share_type_int64_map) Key(key int) Share_type {
	return NewShare_type(delta.CopyAndAppendKeyInt64(b.location, int64(key)))
}
func (b Share_type_int64_map) Rename(from, to int) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyInt64(b.location, int64(from)), int64(to))
}
func (b Share_type_int64_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Share_type_int64_map) Set(value map[int64]*Share) *delta.Op {
	return delta.Set(b.location, value)
}

type Share_type_uint32_map struct {
	location []*delta.Locator
}

func (b Share_type_uint32_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShare_type_uint32_map(l []*delta.Locator) Share_type_uint32_map {
	return Share_type_uint32_map{location: l}
}
func (b Share_type_uint32_map) Key(key int) Share_type {
	return NewShare_type(delta.CopyAndAppendKeyUint32(b.location, uint32(key)))
}
func (b Share_type_uint32_map) Rename(from, to int) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyUint32(b.location, uint32(from)), uint32(to))
}
func (b Share_type_uint32_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Share_type_uint32_map) Set(value map[uint32]*Share) *delta.Op {
	return delta.Set(b.location, value)
}

type Share_type_uint64_map struct {
	location []*delta.Locator
}

func (b Share_type_uint64_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShare_type_uint64_map(l []*delta.Locator) Share_type_uint64_map {
	return Share_type_uint64_map{location: l}
}
func (b Share_type_uint64_map) Key(key int) Share_type {
	return NewShare_type(delta.CopyAndAppendKeyUint64(b.location, uint64(key)))
}
func (b Share_type_uint64_map) Rename(from, to int) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyUint64(b.location, uint64(from)), uint64(to))
}
func (b Share_type_uint64_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Share_type_uint64_map) Set(value map[uint64]*Share) *delta.Op {
	return delta.Set(b.location, value)
}

type Share_type_string_map struct {
	location []*delta.Locator
}

func (b Share_type_string_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShare_type_string_map(l []*delta.Locator) Share_type_string_map {
	return Share_type_string_map{location: l}
}
func (b Share_type_string_map) Key(key string) Share_type {
	return NewShare_type(delta.CopyAndAppendKeyString(b.location, key))
}
func (b Share_type_string_map) Rename(from, to string) *delta.Op {
	return delta.Rename(delta.CopyAndAppendKeyString(b.location, from), to)
}
func (b Share_type_string_map) Delete() *delta.Op {
	return delta.Delete(b.location)
}
func (b Share_type_string_map) Set(value map[string]*Share) *delta.Op {
	return delta.Set(b.location, value)
}

File: lib/pb/groupshare/data/data.op.dart
import 'package:protod/delta.dart' as delta;
import 'package:protod/delta.pb.dart' as delta;
import 'package:fixnum/fixnum.dart' as fixnum;
import 'package:groupshare/pb/groupshare/data/foo/foo.op.dart' as groupshare_data_foo_foo;
import 'package:groupshare/pb/groupshare/data/share.pb.dart' as pb;

Op_root_type Op() {
  return Op_root_type();
}

class Op_root_type {
  Op_root_type();
  Share_type Share() {
    return Share_type([]);
  }
}

class Share_type extends delta.Location {
  Share_type(List<delta.Locator> location) : super(location);
  delta.String_scalar Id() {
    return delta.String_scalar(delta.copyAndAppendField(location, "id", 1));
  }
  delta.String_scalar Name() {
    return delta.String_scalar(delta.copyAndAppendField(location, "name", 2));
  }
  groupshare_data_foo_foo.Foo_type Foo() {
    return groupshare_data_foo_foo.Foo_type(delta.copyAndAppendField(location, "foo", 3));
  }
  delta.Op Delete() {
    return delta.delete(location);
  }

  delta.Op Set(pb.Share value) {
    return delta.set(location, value);
  }

}

class Share_type_list extends delta.Location {
  Share_type_list(List<delta.Locator> location) : super(location);
  Share_type Index(int i) {
    return Share_type(delta.copyAndAppendIndex(location, fixnum.Int64(i)));
  }
  delta.Op Insert(int index, pb.Share value) {
    return delta.insert(delta.copyAndAppendIndex(location, fixnum.Int64(index)), value);
  }

  delta.Op Move(int from, int to) {
    return delta.moveList(delta.copyAndAppendIndex(location, fixnum.Int64(from)), fixnum.Int64(to));
  }

  delta.Op Delete() {
    return delta.delete(location);
  }

  delta.Op Set(List<pb.Share> value) {
    return delta.set(location, value);
  }

}

class Share_type_bool_map extends delta.Location {
  Share_type_bool_map(List<delta.Locator> location) : super(location);
  Share_type Key(bool key) {
    return Share_type(delta.copyAndAppendKeyBool(location, key));
  }
  delta.Op Rename(bool from, bool to) {
    return delta.moveMap(delta.copyAndAppendKeyBool(location, from), delta.keyBool(to));
  }

  delta.Op Delete() {
    return delta.delete(location);
  }

  delta.Op Set(Map<bool, pb.Share> value) {
    return delta.set(location, value);
  }

}

class Share_type_int32_map extends delta.Location {
  Share_type_int32_map(List<delta.Locator> location) : super(location);
  Share_type Key(int key) {
    return Share_type(delta.copyAndAppendKeyInt32(location, key));
  }
  delta.Op Rename(int from, int to) {
    return delta.moveMap(delta.copyAndAppendKeyInt32(location, from), delta.keyInt32(to));
  }

  delta.Op Delete() {
    return delta.delete(location);
  }

  delta.Op Set(Map<int, pb.Share> value) {
    return delta.set(location, value);
  }

}

class Share_type_int64_map extends delta.Location {
  Share_type_int64_map(List<delta.Locator> location) : super(location);
  Share_type Key(int key) {
    return Share_type(delta.copyAndAppendKeyInt64(location, fixnum.Int64(key)));
  }
  delta.Op Rename(int from, int to) {
    return delta.moveMap(delta.copyAndAppendKeyInt64(location, fixnum.Int64(from)), delta.keyInt64(fixnum.Int64(to)));
  }

  delta.Op Delete() {
    return delta.delete(location);
  }

  delta.Op Set(Map<fixnum.Int64, pb.Share> value) {
    return delta.set(location, value);
  }

}

class Share_type_uint32_map extends delta.Location {
  Share_type_uint32_map(List<delta.Locator> location) : super(location);
  Share_type Key(int key) {
    return Share_type(delta.copyAndAppendKeyUint32(location, key));
  }
  delta.Op Rename(int from, int to) {
    return delta.moveMap(delta.copyAndAppendKeyUint32(location, from), delta.keyUint32(to));
  }

  delta.Op Delete() {
    return delta.delete(location);
  }

  delta.Op Set(Map<int, pb.Share> value) {
    return delta.set(location, value);
  }

}

class Share_type_uint64_map extends delta.Location {
  Share_type_uint64_map(List<delta.Locator> location) : super(location);
  Share_type Key(int key) {
    return Share_type(delta.copyAndAppendKeyUint64(location, fixnum.Int64(key)));
  }
  delta.Op Rename(int from, int to) {
    return delta.moveMap(delta.copyAndAppendKeyUint64(location, fixnum.Int64(from)), delta.keyUint64(fixnum.Int64(to)));
  }

  delta.Op Delete() {
    return delta.delete(location);
  }

  delta.Op Set(Map<fixnum.Int64, pb.Share> value) {
    return delta.set(location, value);
  }

}

class Share_type_string_map extends delta.Location {
  Share_type_string_map(List<delta.Locator> location) : super(location);
  Share_type Key(String key) {
    return Share_type(delta.copyAndAppendKeyString(location, key));
  }
  delta.Op Rename(String from, String to) {
    return delta.moveMap(delta.copyAndAppendKeyString(location, from), delta.keyString(to));
  }

  delta.Op Delete() {
    return delta.delete(location);
  }

  delta.Op Set(Map<String, pb.Share> value) {
    return delta.set(location, value);
  }

}

